compiler/src/typed/translprim.re

```diff
@@ -1475,43 +1473,12 @@ let transl_prim = (env, desc) => {
     pla_loc: Location.dummy_loc,
   };
 
-  // `attrs` are attributes which should be applied to the `let` which gets implicitly generated.
-  //
-  // Specifically, consider:
-  //
-  // primitive equal: (a, a) -> Bool = "@equal"
-  //
-  // This will get pseudo-generated into something like (note: not syntactically correct):
-  //
-  // let equal = @disableGC (a, b) => <@equal>(a, b)
-  //
-  // For *constant* values, this is not good enough. Consider:
-  //
-  // primitive heapBase = "@heap.base"
-  //
-  // Since the RHS is just the constant value (inlined here), this binding effectively becomes
-  //
-  // let heapBase = 0x400n
-  //
-  // which means that we need to put the `@disableGC` on the *outside* of the `heapBase` binding.
-  // This is what the `attrs` list here controls; by returning `attrs == [Typedtree.Disable_gc]`,
-  // we will generate
-  //
-  // @disableGC let heapBase = 0x400n
-  //
-  // Which can pass through the rest of the compilation without issue.
-  //
   let (value, typ) =
     switch (prim) {
     | PrimitiveConstant(const) =>
       let (value, typ, attributes) =
         switch (const) {
         // [NOTE] should be kept in sync with `runtime_heap_ptr` and friends in `compcore.re`
-        | HeapBase => (
-            Constant.wasmi32(string_of_int(active_memory_base())),
-            Builtin_types.type_wasmi32,
-            disable_gc,
-          )
         | HeapTypeMetadata => (
             Constant.wasmi32(string_of_int(active_memory_base() + 0x8)),
```

Comment:
In compcore, you removed one instance of `0x8` when working with HeapTypeMetadata, but not here. Why?

Reply:
That one was related to the old static pointer.